src/features/timebank/timebank-api.ts

```diff
@@ -0,0 +1,110 @@
+import { WeeklyCombinedData } from "@functions/schema";
+import { Person, DailyEntriesApi, PersonsApi, DailyEntry, Timespan, PersonTotalTime } from "src/generated/client/api";
+
+/**
+ * Namespace for Timebank API
+ */
+namespace TimebankApi {
+  export const personsClient = new PersonsApi(process.env.TIMEBANK_BASE_URL);
+  export const dailyEntriesClient = new DailyEntriesApi(process.env.TIMEBANK_BASE_URL);
+
+  /**
+   * Get list of timebank users from TimeBank API
+   *
+   * @returns valid persons data
+   */
+  export const getTimebankUsers = async (accessToken: string): Promise<Person[]> => {
+    try {
+      const request = await personsClient.listPersons(true, {
+        headers:
+          { "Authorization": `Bearer ${accessToken}` }
+      });
+      
+      if (request.response.statusCode === 200) {
+        return request.body;
+      }
+
+      throw new Error("Error while loading Persons from Timebank");
+    } catch (error) {
+      console.error(error.toString());
+    }
+  };
+
+  /**
+   * Get time entries for specific timebank user
+   *
+   * @param id from persons data
+   * @param before string for timebank dates
+   * @param after string for timebank dates
+   * @returns Array of time entries for person
+   */
+  export const getDailyEntries = async (id: number, before: string, after: string, accessToken: string): Promise<DailyEntry> => {
+    try {
+      if (id === null) throw new Error("Invalid ID was given (expecting a number)");
+      const request = await dailyEntriesClient.listDailyEntries(id, before, after, undefined, {
+        headers:
+          { "Authorization": `Bearer ${accessToken}` }
+      });
+
+      if (request.response.statusCode === 200) {
+        return request.body[0];
+      }
+      
+      throw new Error(`Error while loading DailyEntries for person ${id} from Timebank`);
+    } catch (error) {
+      console.error(error.toString());
+    }
+  };
+
+  /**
+   * Get totals time entries for user for specific time period
+   *
+   * @param timePeriod of time data to sum
+   * @param person person data from timebank
+   * @param year of data to request
+   * @param week of data to request
```

Comment:
missing month and access token TSDoc @param comments

Reply:
Done